x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/index/store/SearchableSnapshotDirectoryTests.java

```diff
@@ -629,6 +627,8 @@ protected void assertSnapshotOrGenericThread() {
             } finally {
                 Releasables.close(releasables);
             }
+        } finally {
+            terminate(threadPool);
```

Comment:
Gotta make sure to only terminate the pool after shutting down everything else, otherwise we might leak a file listener from an action that has a reference to the file never executing.
See https://gradle-enterprise.elastic.co/s/wccywnq4vwa7i/console-log?anchor=8057